test-framework/core/src/main/java/org/keycloak/test/framework/realm/ManagedRealm.java

```diff
@@ -27,4 +30,37 @@ public RealmResource admin() {
         return realmResource;
     }
 
+    public void updateWithCleanup(RealmUpdate... updates) {
+        RealmRepresentation rep = admin().toRepresentation();
+        cleanup().resetToOriginalRepresentation(rep);
+
+        RealmConfigBuilder configBuilder = RealmConfigBuilder.update(rep);
+        for (RealmUpdate update : updates) {
+            configBuilder = update.update(configBuilder);
+        }
+
+        admin().update(configBuilder.build());
+    }
+
+    public ManagedRealmCleanup cleanup() {
+        if (cleanup == null) {
+            cleanup = new ManagedRealmCleanup();
+        }
+        return cleanup;
+    }
+
+    @Override
+    public void runCleanup() {
+        if (cleanup != null) {
+            cleanup.runCleanupTasks(realmResource);
+            cleanup = null;
+        }
+    }
+
+    public interface RealmUpdate {
+
+        RealmConfigBuilder update(RealmConfigBuilder realm);
```

Comment:
I am missing some implementation of it :-)

Reply:
CleanupTest uses this